### PR TITLE
Add simdjson

### PR DESCRIPTION
**Pull request by elbaro**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/simdjson/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
```
Comment (nehaljwani) [inline recipes/simdjson/build.sh:2]: The project supports CMake: https://github.com/simdjson/simdjson/blob/v0.3.1/CMakeLists.txt

Why not use that?

Comment (elbaro) [reply recipes/simdjson/build.sh:2]: simdjson does not have any documentation about its cmake build:
https://github.com/simdjson/simdjson/issues/820
so I did not assume it is safe to use.

The officially supported usage is using `singleheader/simdjson.{cpp,simdjson.h}`

**Comment by elbaro** [comment]
@conda-forge/staged-recipes  this is ready for review

File: recipes/simdjson/build.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+pushd singleheader
+# shared build
+$CXX --std=c++17 -O2 -fPIC -Wall -Wextra simdjson.cpp -shared -o libsimdjson.so
+# static build
+$CXX --std=c++17 -O2 -Wall -Wextra simdjson.cpp -c -o simdjson.o
+$AR -rc libsimdjson.a simdjson.o
```
Comment (xhochy) [inline recipes/simdjson/build.sh:9]: Please only install the dynamic library.

File: recipes/simdjson/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -e
+
+pushd singleheader
+# shared build
+$CXX --std=c++17 -O2 -fPIC -Wall -Wextra simdjson.cpp -shared -o libsimdjson.so
```
Comment (xhochy) [inline recipes/simdjson/build.sh:6]: Please use $CXXFLAGS, otherwise you might build for the wrong arch.

File: recipes/simdjson/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/simdjson/simdjson/archive/v{{ version }}.tar.gz
+  sha256: 9ae0b77139d6b854fe4d25db5e525762d3f7880f7146329767078397b6ed2f94
+
+build:
+  number: 0
+  skip: true  # [win]
```
Comment (xhochy) [inline recipes/simdjson/meta.yaml:14]: This needs a `run_exports`

**Comment by elbaro** [comment]
@conda-forge/staged-recipes @xhochy addressed comments.

**Pull request merged**